compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -2827,13 +2728,7 @@ impl<'a> Parser<'a> {
     fn parse_try_block(&mut self, span_lo: Span) -> PResult<'a, P<Expr>> {
         let (attrs, body) = self.parse_inner_attrs_and_block()?;
         if self.eat_keyword(kw::Catch) {
-            let mut error = self.struct_span_err(
-                self.prev_token.span,
-                "keyword `catch` cannot follow a `try` block",
-            );
-            error.help("try using `match` on the result of the `try` block instead");
-            error.emit();
-            Err(error)
```

Comment:
I'm not sure about this one - it had both an `emit()` and return the error in an `Err`. I tried doing both `emit_err` and returning the error, but that resulted in it being printed twice, so I left just the return.

Reply:
This was likely a pre-existing bug.